modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/trigger/messaging/ObjectActionsTriggerMessageListener.java

```diff
@@ -38,26 +39,58 @@
  * @author Marco Leov
  */
 @Component(
-	property = "destination.name=" + DestinationNames.OBJECT_ENTRY_ATTACHMENT_DOWNLOAD,
+	property = {
+		"destination.name=" + DestinationNames.OBJECT_ENTRY_ATTACHMENT_DOWNLOAD,
+		"destination.name=" + DestinationNames.OBJECT_ENTRY_ON_AFTER_CREATE,
+		"destination.name=" + DestinationNames.OBJECT_ENTRY_ON_AFTER_UPDATE
+	},
 	service = MessageListener.class
 )
-public class ObjectActionDownloadTriggerMessageListener
-	extends BaseMessageListener {
+public class ObjectActionsTriggerMessageListener extends BaseMessageListener {
 
 	@Activate
 	protected void activate(BundleContext bundleContext) {
-		DestinationConfiguration destinationConfiguration =
+		DestinationConfiguration afterCreateDestinationConfiguration =
+			new DestinationConfiguration(
+				DestinationConfiguration.DESTINATION_TYPE_SYNCHRONOUS,
+				DestinationNames.OBJECT_ENTRY_ON_AFTER_CREATE);
+
+		Destination afterCreationDestination =
+			_destinationFactory.createDestination(
+				afterCreateDestinationConfiguration);
+
+		_serviceRegistration = bundleContext.registerService(
+			Destination.class, afterCreationDestination,
+			MapUtil.singletonDictionary(
+				"destination.name", afterCreationDestination.getName()));
```

Comment:
This code is repeating. I noticed the only thing changing is the second parameter in `new DestinationConfiguration()`.
Can you reuse the code please?

Reply:
Since you are registering more than one, I believe the `deactivate` method will not work as it is.
You will need to adapt. I saw this [example](https://github.com/liferay/liferay-portal/blob/master/modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/configuration/admin/category/FeatureFlagConfigurationCategory.java#L76-L81) in the portal; I think it can help you.